.github/actions/e2e/action.yaml

```diff
@@ -180,3 +175,46 @@ runs:
       fi
       echo "::endgroup::"
     shell: bash
+  - name: Confirm node selection
+    if: always()
+    run: |
+      # Confirm node selection
+      echo "::group::Confirm node selection"
+      success=true
+      expectedNode="armada-worker2"
+
+      for pod in default.armada-*-0.json
+      do
+        echo "pod ${pod/%.json/}"
+        nodeSelector=$(jq -r '.spec.nodeSelector."armada-spark"' "$pod")
+        actualNode=$(jq -r .spec.nodeName "$pod")
+
+        if [[ "$nodeSelector" == "true" ]]
+        then
+          echo "- Node selector exists"
+        else
+          echo "- Node selector is missing"
+          success=false
+        fi
+
+        if [[ "$actualNode" == "$expectedNode" ]]
+        then
+          echo "- Pod assigned to expected node"
+        else
+          echo "- Pod assigned to unexpected node: $actualNode"
+          success=false
+        fi
+        jq '.spec | { "nodeName": .nodeName, nodeSelector: .nodeSelector }' "$pod"
+        echo
+      done
+      "$success"
```

Comment:
this is an interesting way to exit.  I haven't seen that approach before.

Reply:
Yes, nice thing is that all `true` and `false` are doing is `exit 0` or `exit 1`, and `$success` executes those shell commands.